queries/roc/highlights.scm

```diff
@@ -0,0 +1,194 @@
+;---Most generic types---
+(module) @module
+
+(identifier) @variable
+
+(concrete_type) @type
+
+;---annotations----
+(annotation_type_def
+  (annotation_pre_colon
+    (identifier) @type))
+
+(annotation_type_def
+  (annotation_pre_colon
+    (identifier) @function)
+  (function_type))
+
+;----decleration types----
+(value_declaration
+  (decl_left
+    (identifier_pattern
+      (identifier) @variable.parameter)))
+
+;---records----
+(field_name) @variable.member
+
+(record_field_pattern
+  (_
+    (identifier) @variable))
+
+;matches the second identifier and all subsequent ones
+(field_access_expr
+  (identifier) @variable.member)
+
+;----comments----
+[
+  (line_comment)
+  (doc_comment)
+] @comment
+
+;-----Punctuation----
+[
+  "?"
+  (arrow)
+  (back_arrow)
+  (backslash)
+] @punctuation.delimiter
+
+[
+  ","
+  ":"
+] @punctuation.delimiter
+
+[
+  "("
+  ")"
+  "{"
+  "}"
+  "["
+  "]"
+] @punctuation.bracket
+
+[
+  "|"
+  "&"
+  (operator)
+  (wildcard_pattern)
+] @operator
+
+[
+  "if"
+  "then"
+  "else"
+] @keyword.conditional
+
+[
+  (implements)
+  (when)
+  (is)
+  "as"
+  (to)
+] @keyword
+
+;----headers-----
+(interface_header
+  (name) @type.definition)
+
+(imports
+  (imports_entry
+    (module) @module))
+
+(packages
+  (record_pattern
+    (record_field_pattern
+      (field_name) @module)))
+
+(app_name) @string
+
+(import_path) @string
+
+[
+  "app"
+  "packages"
+  "imports"
```

Comment:
Maybe merge this list with the previous list that also targets `@keyword`. Also should `"imports"` be instead highlighted as `@keyword.import`?

Reply:
Here is an example of the use of `imports` in  a simple Roc program:

```
app "hello"
    packages { pf: "https://github.com/roc-lang/basic-cli/releases/download/0.8.1/x8URkvfyi9I0QhmVG98roKBUs_AZRkLFwFJVJ3942YA.tar.br" }
    imports [pf.Stdout]
    provides [main] to pf

main =
    Stdout.line "I'm a Roc application!"
 ```
 
 My understanding is that, as part of the module header, it should be `@module` rather than `@keyword.imports` which I would associate for example with the Go style of importing. If I am wrong, please let me know.

Reply:
the `imports` is a keyword, so should be `@keyword.import`. The `[pf.Stdout]` would be the `@module` (assuming I understand correctly).

Reply:
While you're at it, is it easy / would it make sense to highlight the tarball string as `@string.special.url`?

Reply:
> the `imports` is a keyword, so should be `@keyword.import`. The `[pf.Stdout]` would be the `@module` (assuming I understand correctly).

In Roc, the whole top block in my example  is known as a module header. Every `.roc`  file is a module, and there are different types of modules. We know this particular one is an application module because it begins with the `app` keyword.

Reply:
So should the whole block be highlighted in a single color?

Reply:
> While you're at it, is it easy / would it make sense to highlight the tarball string as `@string.special.url`?

In Roc, packages may be loaded either via URL or from a local filesystem path. So I am not sure it is appropriate to mark that string as a URL.

Reply:
> So should the whole block be highlighted in a single color?

I think different colors could be fine. It is comparable to the usual beginning in Go:

```go
package main

import (
	"fmt"
	"os"
)

// ...
```

In my Neovim setup, `package` and `import` are both colored red; `main` , `(`, and `)` are white; `"fmt"` and `"os"` are green. Nothing per se marks them as "top matter". In Roc, the same could work—or, the whole block could be visually distinct in some way. I am open to whatever seems like it fits in with the norms of the `nvim-treesitter` project.

Reply:
Then `import` is `@keyword.import`, period.

Reply:
(`.import`, not `.import**s**` -- that's exactly why we have that CI job ;))